cmd/skaffold/app/cmd/lsp.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2021 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/signal"
+
+	"github.com/spf13/cobra"
+	"go.lsp.dev/jsonrpc2"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/lint"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/lsp"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/output/log"
+)
+
+var port string
+
+func NewCmdLSP() *cobra.Command {
+	return NewCmd("lsp").
+		WithDescription("Starts skaffold LSP language server that provides lint suggestions to IDEs").
+		WithCommonFlags().
+		WithFlags([]*Flag{
+			{Value: &port, Name: "port", DefValue: ":4389",
+				Usage: "port that the skaffold lsp will expose for connections, requires the form ':<port-number>'. ':4389' by default"}}).
+		Hidden().
+		NoArgs(doLSP)
+}
+
+func doLSP(ctx context.Context, out io.Writer) error {
+	lint.LintRunnerOpts = lint.NewRunnerOpts(ctx, out, opts, createRunner)
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
+	go func() {
+		for sig := range c {
+			if sig == os.Interrupt {
+				os.Exit(0)
+			}
+		}
+	}()
+	// TODO(aaron-prindle) plumb context based termination as well on os.Kill and os.Interrupt
+
+	listen := func(port string) (*net.Listener, error) {
+		listener, err := net.Listen("tcp", port)
+		if err != nil {
+			log.Entry(ctx).Fatalf("Could not bind to address %s: %v", port, err)
+			return nil, err
+		}
+		if os.Getenv("TLS_CERT") != "" && os.Getenv("TLS_KEY") != "" {
+			cert, err := tls.X509KeyPair([]byte(os.Getenv("TLS_CERT")), []byte(os.Getenv("TLS_KEY")))
+			if err != nil {
+				return nil, err
+			}
+			listener = tls.NewListener(listener, &tls.Config{
+				Certificates: []tls.Certificate{cert},
+			})
+		}
+		return &listener, nil
+	}
+	lis, err := listen(port)
+	if err != nil {
+		log.Entry(ctx).Fatalf("encountered error attempting to listen via tcp on port %s: %v", port, err)
+	}
+	defer (*lis).Close()
+
+	fmt.Fprintf(lint.LintRunnerOpts.Out, "skaffold lsp listening for TCP connections on: %s\n", port)
+	connectionCount := 0
+	for {
+		conn, err := (*lis).Accept()
+		if err != nil {
+			log.Entry(ctx).Fatalf("encountered error attempting to accept tcp connection: %v", err)
+		}
+		connectionCount++
+		connectionID := connectionCount
```

Comment:
does `connectionID` get used anywhere besides the logging call? If not should we just remove it and log `connectionCount`?

Reply:
you're right it was used by no longer.  Removed connectionID and use connectionCount now